plugins/woocommerce/src/Admin/PluginsHelper.php

```diff
@@ -600,4 +626,272 @@ public static function maybe_enqueue_scripts_for_connect_notice() {
+			'button_text' => __( 'Enable auto-renewal', 'woocommerce' ),
+			'button_link' => self::WOO_SUBSCRIPTION_PAGE_URL,
+		);
+	}
+
+	/**
+	 * Get formatted notice information for expired subscription.
+	 *
+	 * @param boolean $allowed_link whether the notice description should include a link.
+	 * @return array notice information.
+	 */
+	public static function get_expired_subscription_notice( $allowed_link = true ) {
+		if ( ! WC_Helper::is_site_connected() ) {
+			return array();
+		}
+
+		$is_notice_dismiss = get_user_meta( get_current_user_id(), self::DISMISS_EXPIRED_SUBS_NOTICE, true );
+		if ( ! empty( $is_notice_dismiss ) ) {
+			return array();
+		}
+
+		$subscriptions         = WC_Helper::get_subscriptions();
+		$expired_subscriptions = array_filter(
+			$subscriptions,
+			function ( $sub ) {
+				return $sub['expired'] && ! $sub['lifetime'];
+			},
+		);
+
+		if ( ! $expired_subscriptions ) {
+			return array();
+		}
+
+		$total_expired_subscriptions         = count( $expired_subscriptions );
+		self::$can_show_expiring_subs_notice = false;
+
+		$message = self::get_subscriptions_message(
+			$subscriptions,
+			$expired_subscriptions,
+			$total_expired_subscriptions,
+			array(
+				/* translators: 1) product name 2) URL to My Subscriptions page 3) Product price */
+				'single_manage'           => __( 'Your subscription for <strong>%1$s</strong> expired. <a href="%3$s" >Renew for %4$s</a> to continue receiving updates and streamlined support', 'woocommerce' ),
+				/* translators: 1) product name 2) URL to My Subscriptions page 3) Product price */
+				'multiple_manage'         => __( 'One of your subscriptions for <strong>%1$s</strong> has expired. <a href="%3$s">Renew for %4$s.</a> to continue receiving updates and streamlined support', 'woocommerce' ),
+				/* translators: 1) total expired subscriptions 2) URL to My Subscriptions page */
+				'different_subscriptions' => __( 'You have <strong>%1$s Woo extension subscriptions</strong> that expired. <a href="%2$s">Renew</a> to continue receiving updates and streamlined support', 'woocommerce' ),
+			),
+		);
+
+		return array(
+			'description' => $allowed_link ? $message : preg_replace( '#<a.*?>(.*?)</a>#i', '\1', $message ),
+			'button_text' => __( 'Renew', 'woocommerce' ),
+			'button_link' => self::WOO_SUBSCRIPTION_PAGE_URL,
+		);
+	}
+
+	/**
+	 * Enqueue scripts for woo subscription notice.
+	 *
+	 * @return void
+	 */
+	public static function maybe_enqueue_scripts_for_subscription_notice() {
+		if ( 'woocommerce_page_wc-settings' !== get_current_screen()->id ) {
+			return;
+		}
+
+		WCAdminAssets::register_script( 'wp-admin-scripts', 'woo-subscriptions-notice' );
+		wp_enqueue_script( 'woo-subscriptions-notice' );
+	}
+
+	/**
+	 * Register the routes for  dismissal subscriptions notice.
+	 */
+	public static function woo_subscriptions_notice_dismiss_api() {
+		register_rest_route(
```

Comment:
What do you think about putting the rest route API registration in the `plugins/woocommerce/src/Admin/API` instead? because it seems like it is more suitable place there

Reply:
Addressed [e5f740a](https://github.com/woocommerce/woocommerce/pull/47004/commits/e5f740a38c33b53bc0c67632163909af3029a6a6)